### ISSUE TITLE
Resetting WiFi driver may reset SPI

### ISSUE THREAD

**Issue by daneisinger** [hpssjellis/everything-nrf52840-usb-dongles#2]
I was trying to use Klaus_K’s technique for switching from BLE to Wifi (included in this repository) alongside the SD card support on the MKR ENV shield and found that whenever the wifi driver was de-initialized and re-initialized I was no longer able to access the SD card. I had to call SD.end() after turning off BLE and then SD.begin() after resetting the wifi driver to work around this. Since both the SD card reader/writer and WIFININA are on the SPI bus I suspect that resetting the wifi driver must somehow impact the bus. At the very least it messes with an existing SD card library initialization. Not sure if there’s way to fix this but thought I would share for anyone else who might run into this and get very frustrated.

**Comment by hpssjellis** [comment]
@daneisinger good catch. I haven't used my NRF52840 Dongles for a few years. I hope this helps someone. Perhaps somehow the SS pin can be selected for each peripheral so that they don't mess up their communication.
